net/netcheck/netcheck.go

```diff
@@ -113,6 +114,10 @@ type Report struct {
 	GlobalV4 string // ip:port of global IPv4
 	GlobalV6 string // [ip]:port of global IPv6
 
+	// CaptivePortal is set when we think there's a captive portal that is
```

Comment:
so despite it being a tristate, we're only using two of its states: unset and true? Why not a bool then?

Reply:
We only check if it's `true` to change behaviour, but all three states are valid:

* `unset` for incremental netchecks
* `true` (during non-incremental netchecks) "made request, got non-204"
* `false` (during non-incremental netchecks) for "made request, got 204"

Reply:
The wording of this comment is "is set when we think there's a captive portal"

I guess what does "is set" mean? I interpret it to mean a non-empty string opt.Bool: so either "true" or "false".

But "false" doesn't mean "we think there's a captive portal"

Reply:
Ah, gotcha; update the comment in-code to be more clear.